### PR TITLE
jrnl: 1.9.8 -> 2.4.2

### PR DESCRIPTION
**Pull request by Br1ght0ne**
###### Motivation for this change
https://github.com/jrnl-org/jrnl/releases/tag/v2.4.2

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/python-modules/asteval/default.nix [hunk 1]
```diff
@@ -0,0 +1,30 @@
+{ lib
+, buildPythonPackage
+, fetchPypi
+, pytest
+}:
+
+buildPythonPackage rec {
+  pname = "asteval";
+  version = "0.9.18";
```
Comment (jonringer) [inline pkgs/development/python-modules/asteval/default.nix:9]:
```suggestion
  version = "0.9.18";
  disabled = isPy27;
```

File: pkgs/development/python-modules/asteval/default.nix [hunk 2]
```diff
@@ -0,0 +1,30 @@
+{ lib
+, buildPythonPackage
+, fetchPypi
```
Comment (jonringer) [inline pkgs/development/python-modules/asteval/default.nix:3]:
```suggestion
, fetchPypi
, isPy27
```

**Review by jonringer — COMMENTED** [review]
```
https://github.com/NixOS/nixpkgs/pull/87793
1 package failed to build:
python27Packages.asteval

3 packages built:
jrnl python37Packages.asteval python38Packages.asteval
```

**Comment by Gaelan** [comment]
Build fails on Darwin. Adding `cryptography` to `propagatedBuildInputs` fixes it.

**Comment by filalex77** [comment]
@GrahamcOfBorg build jrnl

**Comment by fabaff** [comment]
Was superseded by #112684.

**Pull request closed without merging**